src/js/components/toolbar/MarkDropPlace.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import {State} from '../../store';
+import {MarkDraggingStateRecord} from '../../store/factory/Inspector';
+import { LyraMarkType, Mark } from '../../store/factory/Mark';
+import {addMark, addGroup } from '../../actions/markActions';
+import {removePlaceHolder} from '../../actions/layoutActions';
+import {getClosestGroupId} from '../../util/hierarchy';
+import { LayoutRecord} from '../../store/factory/Layout';
+
+const imutils = require('../../util/immutable-utils');
+const getInVis = imutils.getInVis;
```

Comment:
we're trying to phase out `getInVis(...)` in favor of `state.getIn(['vis', 'present', ... ])` going forward